src/components/Rating/Rating.js

```diff
@@ -48,18 +50,24 @@ class Rating extends Component {
       iconFilledRenderer
     } = this.props
 
+    const iconProps = { className: classes.icon }
+    if (disabled) {
+      iconProps.nativeColor = grey[300]
+    }
+
     switch (type) {
       case 'normal':
         return iconNormalRenderer
           ? iconNormalRenderer({ ...this.props, index })
-          : (disabled ? React.cloneElement(iconNormal, { nativeColor: grey[300] }) : iconNormal)
+          : React.cloneElement(iconNormal, iconProps)
```

Comment:
Due to this (unrelated) change, the element will _always_ be cloned. I don't know if that's a performance issue (it shouldn't be), but is it worth the slightly improved readability? :thinking:

Reply:
I don't remember doing that change, I don't even recognize that line. I'm reverting this now.

Reply:
ohh now I see ... It's not my change. well, since it's an npm package to be consumed by many and developed by few, I would favor perf over readability. However, this is so insignificant, and there are many other optimizations to be made, so ...

I wouldn't think too much about it. Whatever "feels" right to you, go for it.

Regardless, I would break that function down a bit, and replace the nested ternary operators. Those can cause headaches you know ;)

Reply:
if you want, I can open a separate PR with perf improvements suggestions

Reply:
I can live with _one_ ternary, but the two nested ones are pretty bad, I agree.

Are there so many significant performance improvements? I mean, it's just five buttons after all. :sweat_smile:

Reply:
I don't think any of it is significant. I wanted to point the fact that it **can** be optimized much further, in response to your question:

> I don't know if that's a performance issue (it shouldn't be), but is it worth the slightly improved readability? :thinking: 

I wouldn't worry about it unless someone consuming this package will complain.